src/timestamp.ts

```diff
@@ -61,18 +61,18 @@ export class Timestamp extends LongWithoutOverridesClass {
       if (typeof low.i !== 'number' && (typeof low.i !== 'object' || low.i._bsontype !== 'Int32')) {
         throw new BSONError('Timestamp constructed from { t, i } must provide i as a number');
       }
-      if (low.t < 0) {
+      if (low.t.valueOf() < 0) {
```

Comment:
I do wonder about explicitly calling `.valueOf()`. It is correct, but should we instead respect the type coercion built into JS and use a `Number()` cast? There's a ticket to add [Symbol.toPrimitive](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive) to our BSON types, we would likely want that to be automatically respected here, is what I'm thinking. 

We would have to confirm that `t` and `i` are not `NaN`